### PR TITLE
Shipping Labels: added the endpoint for getting Carriers and Rates

### PR DESCRIPTION
**Pull request by pmusolino**
Part of #4085

## Description
Added the endpoint `POST: wc/v1/connect/label/{{order_id}}/rates` in Networking for fetching the Carriers and Rates passing `siteID`, `orderID`, `originAddress`, `destinationAddress` and `packages`.
The PR is over 500 loc because of the fake JSON response.

## Testing
- Just CI, the endpoint will be used in the next PRs.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Model/ShippingLabel/Packages/Carriers and Rates/ShippingLabelCarriersAndRates.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+/// Represents the rate for a specific shipping carrier
```
Comment (rachelmcr) [inline Networking/Networking/Model/ShippingLabel/Packages/Carriers and Rates/ShippingLabelCarriersAndRates.swift]: nit: This represents all of the available shipping carriers and rates (not a specific one), right?

File: Networking/Networking/Extensions/Array+Woo.swift [hunk 2]
```diff
@@ -18,3 +18,13 @@ extension Array where Element == Int64 {
...
     }
 }
+
+// MARK: - Collection Helpers
+//
+extension Collection {
+    /// Returns the element at the specified index if it is within bounds, otherwise nil.
+    ///
+    subscript (safe index: Index) -> Element? {
+        return indices.contains(index) ? self[index] : nil
```
Comment (rachelmcr) [inline Networking/Networking/Extensions/Array+Woo.swift:28]: This is clever!

**Review by rachelmcr — APPROVED** [review]
Looks good to me! 🚢

**Comment by pmusolino** [comment]
Thank you @rachelmcr for your review! 💯

**Pull request merged**